xbmc/cores/AudioEngine/Sinks/AESinkDirectSound.cpp

```diff
@@ -543,6 +543,7 @@ void CAESinkDirectSound::EnumerateDevicesEx(AEDeviceInfoList &deviceInfoList, bo
     deviceInfo.m_deviceType       = aeDeviceType;
 
     deviceInfo.m_wantsIECPassthrough = true;
+    deviceInfo.m_onlyPCM = true;
```

Comment:
DirectSound was able to do AC3 some years ago. Not true anymore?

Reply:
If you make it: device_type = AE_DEVTYPE_PCM - that should not be needed.

Reply:
I don't know if it's possible use for AC3 but it's certain that it can't be used for TrueHD, DTSHD-MA, etc. So it doesn't make sense to want to use DirectSound for PT when WASAPI is already available for everything.

Reply:
The big advantage of Directsound is: "Soundserver" - people used it back at the time, to use it in combination with mixing and desktop sounds. TrueHD, DTSHD, etc. is not enumerated for DirectSound anways. So people will manually switch to Wasapi in that case anyways.

I would keep the AC3 possibility, as it does not introduce additional complexity.

Reply:
There are only two settings:

audio device
audio PT device

If audioPTdevice = DirectSound (so you can use AC3 assuming it still works) How do you tell Kodi to use WASAPI for everything else? It is not possible with the current settings.

Then people complains "TrueHD PT does not works"  (because wrong Kodi settings is allowed....)

Reply:
For now, if you have selected DirectSound, you did not get the TrueHD option and only AC3 / DTS was settable - so people always have to switch to Wasapi anyways.

Though for coding purposes: if we don't want DirectSound to do PT, it's enough to enumerate it AE_DEVTYPE_PCM and it will be gone from the PT list.

Reply:
> I would keep the AC3 possibility

I have removed this change for now so that DirectSound for AC3 PT can continue to be used. It can always be removed later.